src/fileCompression.ts

```diff
@@ -0,0 +1,120 @@
+            try {
+                // Check for write access to ensure we'll be able to clean it up afterward.
+                await fs.promises.access(filePath, fs.constants.R_OK | fs.constants.W_OK);
+                // The path is accessible, and needs to be compressed.
+                return true;
+            }
+            catch (error) {
+                console.error(`shouldCompressFile: failed to access ${filePath}.`);
+                return false;
+            }
+        };
+    }
+
+    private static async compressFile(filePath: string, compressedDestinationFilePath: string) {
+        const inputFileSize = fs.statSync(filePath).size;
+
+        const outputCompressedFileStream = fs.createWriteStream(compressedDestinationFilePath);
+
+        const brotliCompressStream = zlib.createBrotliCompress({
+            params: {
+              [zlib.constants.BROTLI_PARAM_MODE]: zlib.constants.BROTLI_MODE_TEXT,
+              [zlib.constants.BROTLI_PARAM_QUALITY]: 11,
+              [zlib.constants.BROTLI_PARAM_SIZE_HINT]: inputFileSize,
+            }
+        });
+        await streamPromises.pipeline(fs.createReadStream(filePath), brotliCompressStream, fs.createWriteStream(compressedDestinationFilePath));
+    }
+
+    public static async getDecompressedContents(filePath: string) : Promise<string> {
+        let output = "";
+
+        const pathIsCompressed = path.extname(filePath) === FileCompression.compressedFileExtension;
+        console.log(`File ${filePath} (extension: ${path.extname(filePath)}) is ${!pathIsCompressed ? "not " : ""}compressed`);
+        const onDiskFileStream = fs.createReadStream(filePath);
+        const brotliDecompress = zlib.createBrotliDecompress();
+        let streamToListenTo: stream.Readable = pathIsCompressed ? brotliDecompress : onDiskFileStream;
+
+        const chunks: Uint8Array[] = [];
+        streamToListenTo.on(
+            "data", (chunk) => { chunks.push(Buffer.from(chunk)) }).on(
+            "error", (err) => { console.log(`Failed to read ${filePath}`)}).on(
+            `${pathIsCompressed ? "finish" : "end"}`, () => { output = Buffer.concat(chunks).toString('utf8'); });
+
+        if (pathIsCompressed) {
+            await streamPromises.pipeline(onDiskFileStream, brotliDecompress);
+        }
+        else {
+            await streamPromises.finished(onDiskFileStream);
+        }
+        return output;
+    }
+
+    public static async verifyContentsIdentical(uncompressedFilePath: string, compressedFilePath: string) : Promise<boolean> {
+        const uncompressedFileContents = await this.getDecompressedContents(uncompressedFilePath);
+        if (uncompressedFileContents.length < 100) {
+            // Extra paranoia: confirm the uncompressed file is reasonably large before proceeding.
+            console.error(`verifyContentsIdentical: uncompressed file (${uncompressedFilePath}) smaller than expected (length=${uncompressedFileContents.length})`;

```

Comment:
missing closing paranthesis and won't compile (i can push and fix after testing reparse locally)

Reply:
Oops, forgot to check the `tsc` output.